iTechArt.Service/Parsers/GroceryParsers.cs

```diff
@@ -89,34 +117,51 @@ public async Task RecordExcelToDatabase(IFormFile formFile)
+            return _grocery.ToArray();
         }
         /// <summary>
-        /// Parsing XML format grocery files
+        /// Parsing XML format grocery files.
         /// </summary>
-        public async Task RecordXmlToDatabase(IFormFile formFile)
+        public async Task<IGrocery[]> XmlParseAsync(IFormFile formFile)
         {
             _grocery = new List<IGrocery>();
             XDocument xm = new XDocument();
             using (var reader = new StreamReader(formFile.OpenReadStream()))
             {
             var xdoc = XDocument.Load(reader);
-            reader.ReadToEnd();
+             await reader.ReadToEndAsync();
                 var items = from item in xdoc.Descendants("dataset").Elements("record")
-                            select new GroceryDTO
+                            select new GroceryDTO()
                             {
-                                FirstName = item.Element("first_name").Value,
-                                LastName = item.Element("last_name").Value,
-                                Email = item.Element("email").Value,
-                                Gender = (Gender)Enum.Parse(typeof(Gender), item.Element("gender").Value),
-                                Birthday = (DateTime)item.Element("birthday"),
-                                JobTitle = item.Element("job_Title").Value,
-                                DepartmentRetail = item.Element("department_retail").Value,
-                                Salary = (double)item.Element("salary")
+                                FirstName = item.Element(GroceryIndexConstants.FIRSTNAME).Value,
+                                LastName = item.Element(GroceryIndexConstants.LASTNAME).Value,
+                                Email = item.Element(GroceryIndexConstants.EMAIL).Value,
+                                Gender = Enum.Parse<Gender>(item.Element(GroceryIndexConstants.GENDER).Value),
+                                Birthday = (DateTime)item.Element(GroceryIndexConstants.BIRTHDAY),
+                                JobTitle = item.Element(GroceryIndexConstants.JOBTITLE).Value,
+                                DepartmentRetail = item.Element(GroceryIndexConstants.DEPARTMENTRETAIL).Value,
+                                Salary = (double)item.Element(GroceryIndexConstants.SALARY)
                             };
 
-                _grocery = items.ToList<IGrocery>();
-                await _groceryRepository.AddGroceriesAsync(_grocery);
+                return items.ToArray();
             }
         }
     }
+    /// <summary>
+    /// Mapping item for Csv helper.
+    /// </summary>
+    public sealed class GroceryMap : ClassMap<GroceryDTO>
+    {
+        public GroceryMap()
+        {
+            Map(p => p.FirstName).Name("first_name");
+            Map(p => p.LastName).Name("last_name");
+            Map(p => p.Email).Name("email");
+            Map(p => p.Gender).Name("gender");
+            Map(p => p.Birthday).Name("birthday");
```

Comment:
if gender column in CVS file will have value for Gender column like "abrakadabra" - you can get exception (depends on settings). In you case you must create converter for check non standard situation